packages/remix-dev/compiler/compilerServer.ts

```diff
@@ -108,6 +108,9 @@ const createEsbuildConfig = (
       "process.env.REMIX_DEV_SERVER_WS_PORT": JSON.stringify(
         config.devServerPort
       ),
+      "process.env.REMIX_CONFIG_FEATURES_V2_META": JSON.stringify(
+        config.features.v2_meta
+      ),
```

Comment:
I don't think this covers unbundles CJS use-cases - I threw a `console.log("meta", process.env.REMIX_CONFIG_FEATURES_V2_META);` inside our `Meta` component and it logs `undefined` on the server and `true` on the client with the flag set.

Doing the same thing we do for the dev server port fixed it for my `remix dev` use case: https://github.com/remix-run/remix/blob/main/packages/remix-dev/config.ts#L401

That likely means we'll be mapping feature flag name -> env var name in a few spots so maybe we oughta define them in a new `features.ts` file and import from there wherever needed?

```js
export const features = {
  v2_meta: "REMIX_CONFIG_FEATURES_V2_META"
};
```

Reply:
@brophdawg11 You wouldn't look expect a value in `meta` because it's not going to be defined at runtime. Instead we'd do something like:

```ts
function Meta_v1() { /* ... */ }
function Meta_v2() { /* ... */ }
export const Meta = process.env. REMIX_CONFIG_FEATURES_V2_META === "true" ? Meta_v2 : Meta_v1;
```

Reply:
Why are we using env vars at all? This is surprising to me. The build and the runtime both have access to the remix config, is there a reason we can't just read from there?

Reply:
Seems like we could put this on `window.__remixContext` for the browser instead of hacking it into a fake clientside `process.env`.

Server runtime has access to the config naturally, client gets access through hydration at runtime. Can you explore that instead?

Reply:
I assumed we didn't want to ship both old/new code flows in the client bundle, but if we don't mind that then I'm on board with runtime config reading

Reply:
I think it's ok to support both during the transition. Then at some point, remove the old code and focus on new.

Anyway, the RFC already proposes a `objectToMetaArray` and `mergeRouteMeta` to handle the old syntax. I would make it so that the old code simply uses these helpers to convert to the new syntax under the covers. That way there's only a single code path. 

That's basically what my PR did. It converted object to array, then the rest of the code only used the array.

Reply:
We decided on a runtime check for now. This means we need to add some data to the entry context that we can access in both builds. Updated accordingly.